src/app/upgrade.cpp

```diff
@@ -0,0 +1,376 @@
+
+        libt::entry resumeData(libt::entry::dictionary_t);
+        resumeData = fast;
+
+        resumeData["qBt-savePath"] = query.value(4).toString().toStdString();
+        resumeData["qBt-ratioLimit"] = query.value(5).toInt();
+        resumeData["qBt-seedingTimeLimit"] = query.value(6).toInt();
+        resumeData["qBt-category"] = query.value(7).toString().toStdString();
+        /*???*/ resumeData["qBt-tags"] = query.value(8).toString().toStdString();
+        resumeData["qBt-name"] = query.value(9).toString().toStdString();
+        resumeData["qBt-seedStatus"] = query.value(10).toBool();
+        resumeData["qBt-tempPathDisabled"] = query.value(11).toBool();
+        queuePos = query.value(3).toInt();
+        resumeData["qBt-queuePosition"] = (queuePos + 1); // qBt starts queue at 1. Compatibility with pre 4.1.5 versions
+        resumeData["qBt-hasRootFolder"] = query.value(12).toBool();
+
+        // The 'metadata' var might be a magnet URI saved into the COL_METADATA column before the metadata
+        // were received from the network. Internally addTorrent_impl() checks if the constructed below
+        // MagnetUri is valid.
+        BitTorrent::MagnetUri magnetUri{metadata};
+        if (magnetUri.isValid()) {
+            isMagnetUri = true;
+            resumeData["qBt-magnetUri"] = metadata.toStdString();
+            resumeData["qBt-paused"] = query.value(13).toBool();
+            resumeData["qBt-forced"] = query.value(14).toBool();
+            resumeData["qBt-firstLastPiecePriority"] = query.value(15).toBool();
+            resumeData["qBt-sequential"] = query.value(16).toBool();
+        }
+
+        libt::bencode(std::back_inserter(fastresume), resumeData);
+    }
+
+    // This should be moved back into Session::initResumeFolder()
+    // when the upgrade code is removed
+    void initiliazeDB(const QString &dbPath)
+    {
+        if(!QSqlDatabase::isDriverAvailable("QSQLITE"))
+            throw RuntimeError {QCoreApplication::translate("BitTorrent::Session", "The Qt version being used doesn't have SQLite support.")};
+
+        using namespace BitTorrent::DB;
+        QSqlDatabase db = QSqlDatabase::addDatabase("QSQLITE", MAIN_CONNECTION_NAME);
+        db.setDatabaseName(dbPath);
+
+        if (!db.open())
+            throw RuntimeError {QCoreApplication::translate("BitTorrent::Session", "Cannot open database. Error: %1").arg(db.lastError().text())};
+
+        // Check DB version
+        QSqlQuery query(db);
+        query.exec(QLatin1String("PRAGMA user_version"));
+        if (query.next() && query.value(0).toInt())
+            return;
+
+        // Probably 1st run.
+        // Input PRAGMAs and create table
+        query.exec(QLatin1String("PRAGMA auto_vacuum = FULL"));
+        query.exec(QString("PRAGMA user_version = %1").arg(BitTorrent::DB::VERSION));
+
+        query.exec(QString("CREATE TABLE %1 "
+                           "("
+                           "%2 TEXT PRIMARY KEY NOT NULL, " // hash
```

Comment:
I wondered for a long time whether there were really good reasons to break the readability of SQL queries like this. I've only found two possible reasons, but none of them seems to me to be strong enough:
1. An attempt to protect yourself from possible typos in queries. But it looks ridiculous, because the probability to make a typo in any other query word is not less than in the names of tables/fields. In addition, they all appear at the first test runs.
2. Ability to easily change table/field names in the future. This is also doubtful... It may look useful only in the early stages, but once it reaches the first release (in fact, once it will be merged in master), it will become very difficult to change the database format, so it will become almost unclaimed.

Reply:
Do you mean the inline C++ comments? I did it for readability and better reviewing of the appropriate var types chosen (in the SQL context), since it involves many columns. Then it will be used as quick reference each time we want to write a new SQL statement.

If you mean my general approach in other SQL statements to split them: I just split them in their logical parts to increase readability. And when the line becomes too long.

Reply:
Do you really think that when I say about "break the readability of SQL queries", I mean either comments or structural splitting that should improve the readability?
I meant the fact that you separate all names from SQL queries.

Reply:
Oh you mean the QString substitutions? I did it to prevent typos and to ease refactoring of this PR, in case it is decided to use a different name for eg the table, before the PR is merged.

Reply:
> I did it to prevent typos and to ease refactoring of this PR, in case it is decided to use a different name for eg the table, before the PR is merged.

Then par. 2 (from my comment) is applicable to it:
>Ability to easily change table/field names in the future. This is also doubtful... It may look useful only in the early stages, but once it reaches the first release (in fact, once it will be merged in master), it will become very difficult to change the database format, so it will become almost unclaimed.